src/Sankey.js

```diff
@@ -0,0 +1,280 @@
+    this._nodes = accessor("nodes");
+    this._nodeWidth = 30;
+    this._on.mouseenter = () => {};
+    this._on["mouseleave.shape"] = () => {
+      this.hover(false);
+    };
+    this._on["mousemove.shape"] = (d, i) => {
+      if (this._focus && this._focus === d.id) {
+        this.hover(false);
+        this._on.mouseenter.bind(this)(d, i);
+
+        this._focus = undefined;
+      }
+      else {
+        const id = this._nodeId(d, i),
+              node = this._nodeLookup[id],
+              nodeLookup = Object.keys(this._nodeLookup).reduce((all, item) => {
+                all[this._nodeLookup[item]] = !isNaN(item) ? parseInt(item) : item;
+                return all;
+              }, {});
+
+        const links = this._linkLookup[node];
+        const filterIds = [id];
+
+        links.forEach(l => {
+          filterIds.push(nodeLookup[l]);
+        });
+
+        this.hover((h, x) => {
+          if (h.source && h.target) {
+            return h.source.id === id || h.target.id === id;
+          }
+          else {
+            return filterIds.includes(this._nodeId(h, x));
+          }
+        });
+      }
+    };
+    this._path = sankeyLinkHorizontal();
+    this._sankey = sankey();
+    this._shape = constant("Rect");
+    this._shapeConfig = assign(this._shapeConfig, {
+      Path: {
+        fill: "none",
+        hoverStyle: {
+          "stroke-width": (d, i) => Math.max(1, Math.abs(d.source.y1 - d.source.y0) * (this._value(d, i) / this._value(d.source, i)) - 15)
+        },
+        label: false,
+        stroke: "#DBDBDB",
+        strokeOpacity: 0.2,
+        strokeWidth: (d, i) =>
+          Math.max(1, Math.abs(d.source.y1 - d.source.y0) * (this._value(d, i) / this._value(d.source, i)) - 20)
+      },
+      Rect: {}
+    });
+    this._value = accessor("value");
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const nodes = this._nodes
+      .map((id, i) => {
+        const n = id;
+
+        return {
+          __d3plus__: true,
+          data: n,
+          id: this._nodeId(id, i),
+          node: n,
+          shape: "Rect"
+        };
+      })
+      .filter(n => n);
+
+    const nodeLookup = this._nodeLookup = nodes.reduce((obj, d, i) => {
+      obj[d.id] = i;
+      return obj;
+    }, {});
+
+    const links = this._links.map((link, i) => {
+      const check = ["source", "target"];
+      const linkLookup = check.reduce((result, item) => {
+        result[item] =
+          typeof link[item] === "number"
+            ? nodeLookup[link[item]]
+            : nodeLookup[link[item]];
+        return result;
+      }, {});
+
+      return {
+        source: linkLookup.source,
+        target: linkLookup.target,
+        value: this._value(link, i) || 1
```

Comment:
If we change the default value in the constuctor to this: `this._value = constant(1);`

Then, this line can just be: `value: this._value(link, i)`

Less code is better 😏 

The `constant` function just creates and accessor that returns a constant value, but can be used just like any other accessor.